src/section/section_proof_chain.rs

```diff
@@ -52,30 +53,33 @@ impl SectionProofChain {
         self.tail.push(Block { key, signature })
     }
 
-    pub(crate) fn first_key(&self) -> &bls::PublicKey {
+    /// Returns the first key(head) from the chain
```

Comment:
The head / tail thing is supposed to be an implementation detail. I'd just say "Returns the first key of the chain" here.